driver/src/models/order.rs

```diff
@@ -7,7 +7,6 @@ use crate::models::{Serializable, RollingHashable, iter_hash};
 #[derive(Clone, Deserialize, Eq, Ord, PartialEq, PartialOrd)]
 #[serde(rename_all = "camelCase")]
 pub struct Order {
-    pub slot_index: u32,
```

Comment:
What was the reason for this change? is this being recorded elsewhere? Does it not matter? Should we stop emitting the event in the contract? I think the contract needs to know for the purpose of fees.

Reply:
The field is never read in the contract code. It's required to be stored in the database, because we need to be able to query in a specific order. However, the driver then never has to read this index again since it knows that all orders are for the queried one.

I noticed this when trying to create orders for a standing order batch. There it would have been weird to think about `slot_indexes`. It would be possible to create "fake" slot indices, but given we don't even use this field, I removed it.

Reply:
I feel like the contract is going to need this at some point if the fee model depends on the order index. If it isn't needed here for now then I suppose this was a good find!

Reply:
I just hope we are not relying on the insertion order of the events in the DB, since I am still unsure if we can rely on them to be emitted in the order that they are collected.

Reply:
> I feel like the contract is going to need this at some point if the fee model depends on the order index.

The contract and the db need it, but not the driver. I'd hope that the fee model doesn't affect the driver.

> I just hope we are not relying on the insertion order of the events in the DB

That's a good find, we are indeed relying on that order. I think we can rely on the SC events being processed in order (that's a guarantee from the listener), however we should just add the sort criteria. Ok to do that in another PR? Even today since we never sort by slot_index (it's not accessed anywhere in the code) this potential problem exists.

Reply:
Also, technically (according to the formal specification) there should be no discrimination of an order's index in a given auction, and it is much easier to prove such a fact if the index is never passed to the solver in the first place. So in a sense, this could be considered a benefit.